cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSICKANSink.java

```diff
@@ -364,13 +404,26 @@ public String getResName(boolean enableLowercase) {
         } // getResName
 
         public void initialize(NGSIEvent event) throws CygnusBadConfiguration {
+        	LOGGER.debug("[" + this.getClass() + "] INITIALIZE");
             service = event.getServiceForNaming(enableNameMappings);
             servicePathForData = event.getServicePathForData();
             servicePathForNaming = event.getServicePathForNaming(enableGrouping, enableNameMappings);
             entityForNaming = event.getEntityForNaming(enableGrouping, enableNameMappings, enableEncoding);
-            orgName = buildOrgName(service);
-            pkgName = buildPkgName(service, servicePathForNaming);
-            resName = buildResName(entityForNaming);
+            if(!pkgByEntity) {
+            	orgName = buildOrgName(service);
+            	pkgName = buildPkgName(service, servicePathForNaming);
+            	resName = buildResName(entityForNaming);
+            }
+            else if(pkgByEntity) {
```

Comment:
it's not necessary to add second if

Reply:
Fixed in f1f9737cf4ef5d7a113be681a47790b8d860c0f3